std/polynomial/polynomial.go

```diff
@@ -51,6 +116,43 @@ func (m MultiLin) fold(api frontend.API, at frontend.Variable) {
 	}
 }
 
+func (m *MultiLin) FoldParallel(api frontend.API, r frontend.Variable) utils.Task {
```

Comment:
We do not need it - circuit compilation happen sequentially anyway and you cannot parallelize it.